src/osm_configurator/view/toplevelframes/project_head_frame.py

```diff
@@ -32,8 +59,369 @@ def __init__(self, state_manager, export_controller, project_controller):
+                                                         border_width=button_constants_i.ButtonConstants.BUTTON_BORDER_WIDTH.value,
+                                                         fg_color=button_constants_i.ButtonConstants.BUTTON_FG_COLOR_ACTIVE.value,
+                                                         hover_color=button_constants_i.ButtonConstants.BUTTON_HOVER_COLOR.value,
+                                                         border_color=button_constants_i.ButtonConstants.BUTTON_BORDER_COLOR.value,
+                                                         text_color=button_constants_i.ButtonConstants.BUTTON_TEXT_COLOR.value,
+                                                         command=self._calculate_button_pressed,
+                                                         text="Calculate")
+        self._calculate_button.grid(row=0, column=6, rowspan=2, columnspan=1)
+
+        # Options Button
+
+        # Options Icon Used: https://www.flaticon.com/free-icon/cogwheel_44427
+        options_icon = customtkinter.CTkImage(light_image=Image.open("../view_icons/options.png"),
+                                              dark_image=Image.open("../view_icons/options.png"),
+                                              size=(ICON_HEIGHT_AND_WIDTH, ICON_HEIGHT_AND_WIDTH))
+        self._options_button = customtkinter.CTkButton(master=self, height=BUTTON_HEIGHT, width=BUTTON_WIDTH,
+                                                       corner_radius=button_constants_i.ButtonConstants.BUTTON_CORNER_RADIUS.value,
+                                                       border_width=button_constants_i.ButtonConstants.BUTTON_BORDER_WIDTH.value,
+                                                       fg_color=button_constants_i.ButtonConstants.BUTTON_FG_COLOR_ACTIVE.value,
+                                                       hover_color=button_constants_i.ButtonConstants.BUTTON_HOVER_COLOR.value,
+                                                       border_color=button_constants_i.ButtonConstants.BUTTON_BORDER_COLOR.value,
+                                                       text_color=button_constants_i.ButtonConstants.BUTTON_TEXT_COLOR.value,
+                                                       command=self._options_button_pressed,
+                                                       text="",
+                                                       image=options_icon)
+        self._options_button.grid(row=0, column=7, rowspan=2, columnspan=1)
 
     def activate(self):
-        pass
+        # IF frame is activated, it is unlocked
+        self._locked = False
+
+        # Getting what is the current state
+        current_state: state_i.State = self._state_manager.get_state()
+        current_state_name = current_state.get_state_name()
+
+        # Activating all Buttons first, to prevent all buttons getting disabled eventually
+        self._main_menu_button.configure(state="normal",
```

Comment:
Refactoring by putting buttons in a list and interating over the list. This will stop the code repetitions

Reply:
a list for all buttons has been added